### PR TITLE
jsonplatform.md: Added documentation

### PR DESCRIPTION
**Pull request by Hbrinj**
closes: #557 
Signed-off-by: Houman Brinjcargorabi <hbrinjcar@gmail.com>

### CODE REVIEW

File: docs/jsonplatform.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+libmraa to just 'work' with the unsupported platform. JSON platform **does not work**
+with boards which require advanced functions which overwrite default functionality or
+boards that require the mux functionality.
+
+Syntax
+------
+
+The JSON file that libmraa reads needs specific values to be set by default, they
+are required in every JSON file intended to be used to load the platform.
+The keywords are case sensative so be aware when writing the JSON file.
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: typo: sensative should be sensitive

File: docs/jsonplatform.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+Each heading is a top level key in the JSON file, each keys type is Array
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: typo: keys should be key's

File: docs/jsonplatform.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+|pwmMaxPeriod	|int	|no		| The max PWM period				|
+|pwmMinPeriod	|int	|no		| The min PWM period				|
+
+
+### layout
+
+**THIS INDEX'S THE PIN ARRAY**
+There needs to be a label for each available pin, e.g if pin_count=28 then there should be 28
+pin labels. Each JSON element under layout is index from the beginning this also means because
+of the one-to-one relationship between using a pin that we need to set pin 0 as an invalid pin.
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: That statement is IMHO a bit overloaded and ambiguous because of that, Maybe rephrase?

File: docs/jsonplatform.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+Each set of keys should be enclosed in a json object.
+
+|Key		|Type	|Required	|Description					|
+|---------------|-------|---------------|-----------------------------------------------|
+|label		|string	|yes		| The label used to describe the pin		|
+|invalid	|boolean|no		| Sets the labeled pin as an invalid pin		|
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: closing table border is off

File: docs/jsonplatform.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+## SPI
+
+**THIS INDEX'S THE SPI ARRAY**
+This is used to setup the SPI functionality. The total number of json objects and the index
+cannot exceed the spi_count. Each set of keys should be enclosed in a json object.
+
+|Key		|Type	|Required	|Description					|
+|---------------|-------|---------------|-----------------------------------------------|
+|index		|int	|yes		| The index into the SPI array			|
+|chipID		|int	|yes		| The parent ID
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: No closing table border

File: docs/jsonplatform.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+|clock		|int	|no		| The clock used for transmissions		|
+|miso		|int	|no		| Pin used for incoming data from the slave	|
+|mosi		|int	|no		| Pin used for outgoing data from the master	|
+|chipselect	|int	|no		| Pin used to select the slave device		|
+|default	|boolean|no		| Sets the default SPI device			|
+
+
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: surplus whitespace

File: docs/jsonplatform.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+index cannot exceed the total pin_count and the number of JSON ojects cannot exceed the
+gpio_count total. The index is also used to set the specified pin's capabilities. Each set
+of keys should be enclosed in a json object.
+
+|Key		|Type	|Required	|Description					|
+|---------------|-------|---------------|-----------------------------------------------|
+|index		|int	|yes		| used to index the pin array			|
+|rawpin		|int	|yes		| used to reference the actual IO pin		|
+
+
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: surplus whitespace

File: docs/jsonplatform.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+pin labels. Each JSON element under layout is index from the beginning this also means because
+of the one-to-one relationship between using a pin that we need to set pin 0 as an invalid pin.
+Each set of keys should be enclosed in a json object.
+
+|Key		|Type	|Required	|Description					|
+|---------------|-------|---------------|-----------------------------------------------|
+|label		|string	|yes		| The label used to describe the pin		|
+|invalid	|boolean|no		| Sets the labeled pin as an invalid pin		|
+
+
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: surplus whitespace

File: docs/jsonplatform.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+|gpio_count	|int	|no		| The subset of pincount which able to do GPIO	|
+|i2c_count	|int	|no		| The number of usable I2C ports		|
+|aio_count	|int	|no		| The number of usable AIO ports		|
+|spi_count	|int	|no		| The number of usable SPI ports		|
+|uart_count	|int	|no		| The total UART ports				|
+|pwmDefPeriod	|int	|no		| The default PWM period			|
+|pwmMaxPeriod	|int	|no		| The max PWM period				|
+|pwmMinPeriod	|int	|no		| The min PWM period				|
+
+
```
Comment (alext-mkrs) [inline docs/jsonplatform.md]: surplus whitespace

**Comment by Hbrinj** [comment]
@alext-mkrs Thanks Alex :)

**Comment by alext-mkrs** [comment]
My pleasure :) Thanks for preparing this one.

**Comment by Hbrinj** [comment]
doneski

**Comment by arfoll** [comment]
Merged - thanks for finishing this! - watch your trailing whitespace and tabs. It only lined up in vim because of the tabspacing, it didn't really line up :/
